docs/agent/secrets.md

```diff
@@ -0,0 +1,249 @@
+# [BETA] Secrets Management
+
+**This feature is in beta and its options or behavior might break between
+minor or bugfix releases of the Agent.**
+
+This feature is only available on Linux at the moment.
+
+Starting with version `6.3.0`, the agent is able to leverage the `secrets`
+package in order to call a customer-provided executable to handle retrieval or
```

Comment:
General comment: in these docs I think we should avoid writing `customer` and prefer `user`, so here: `user-provided executable`. Or, alternatively, simply `you`/`your`.